third_party/libunwindstack/tests/PeCoffEpilogTest.cpp

```diff
@@ -0,0 +1,579 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PeCoffEpilog.h"
+
+#include <memory>
+#include <vector>
+
+#include <unwindstack/MachineX86_64.h>
+#include <unwindstack/RegsX86_64.h>
+#include "Check.h"
+#include "MemoryFake.h"
+
+#include <gtest/gtest.h>
+
+namespace unwindstack {
+
+// While XMM registers can occur in epilog code (and as UNWIND_INFO codes), they can
+// not be pushed to the stack, they are always saved with a 'mov' instruction into the
+// area allocated on the stack for the current stack frame. In epilogs, the
+// corresponding restore operations do not exist and we therefore do not have to care
+// about them here.
+enum Register : uint8_t {
+  RAX = 0,
+  RCX = 1,
+  RDX = 2,
+  RBX = 3,
+  RSP = 4,
+  RBP = 5,
+  RSI = 6,
+  RDI = 7,
+  R8 = 8,
+  R9 = 9,
+  R10 = 10,
+  R11 = 11,
+  R12 = 12,
+  R13 = 13,
+  R14 = 14,
+  R15 = 15,
+};
+
+// Only non-volatile registers should be used for these:
+// RBX, RBP, RDI, RSI, RSP, R12, R13, R14, R15
+struct PopOp {
+  Register reg;
+  uint64_t value;
+};
+
+struct EpilogOptions {
```

Comment:
Even this is a test class only, it is a bit unclear what those structs are used for. So maybe add a brief comment.

Reply:
Thanks!